crates/utils/src/tracing/grpc.rs

```diff
@@ -0,0 +1,116 @@
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+/// A [`trace_fn`](tonic::transport::server::Server) implementation for the block producer which
+/// adds open-telemetry information to the span.
+///
+/// Creates an `info` span following the open-telemetry standard: `block-producer.rpc/{method}`.
+/// Additionally also pulls in remote tracing context which allows the server trace to be connected
+/// to the client's origin trace.
+pub fn block_producer_trace_fn(request: &http::Request<()>) -> tracing::Span {
+    let span = if let Some("SubmitProvenTransaction") = request.uri().path().rsplit('/').next() {
+        tracing::info_span!("block-producer.rpc/SubmitProvenTransaction")
+    } else {
+        tracing::info_span!("block-producer.rpc/Unknown")
+    };
+
+    add_otel_span_attributes(span, request)
+}
+
+/// A [`trace_fn`](tonic::transport::server::Server) implementation for the store which adds
+/// open-telemetry information to the span.
+///
+/// Creates an `info` span following the open-telemetry standard: `store.rpc/{method}`. Additionally
+/// also pulls in remote tracing context which allows the server trace to be connected to the
+/// client's origin trace.
+pub fn store_trace_fn(request: &http::Request<()>) -> tracing::Span {
+    let span = match request.uri().path().rsplit('/').next() {
+        Some("ApplyBlock") => tracing::info_span!("store.rpc/ApplyBlock"),
```

Comment:
Open-telemetry standard specifies `{package}.{service}/{method}` for RPC span names. Technically our protobuf says the service is called `Api`.